### PR TITLE
samples: wifi: shell: Make PSA backend for Wi-Fi credentials default

### PR DESCRIPTION
**Pull request by simensrostad**
Make PSA backend for Wi-Fi credentials default for nRF7002 Non-Secure builds.

### CODE REVIEW

File: samples/wifi/shell/boards/nrf7002dk_nrf5340_cpuapp_ns.conf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# Store Wi-Fi credentials using PSA storage
+CONFIG_WIFI_CREDENTIALS_BACKEND_PSA=y
```
[the remark this answers is not in the document]
Comment (simensrostad) [reply samples/wifi/shell/boards/nrf7002dk_nrf5340_cpuapp_ns.conf:8]: Done

File: samples/wifi/shell/overlay-tfm.conf [hunk 2]
```diff
@@ -26,4 +26,4 @@ CONFIG_NET_RX_STACK_SIZE=2048
 CONFIG_NET_MGMT_EVENT_QUEUE_TIMEOUT=5000
 
 CONFIG_MBEDTLS_PSA_CRYPTO_C=y
-CONFIG_MBEDTLS_HEAP_SIZE=1024
+CONFIG_MBEDTLS_HEAP_SIZE=16384
```
Comment (krish2718) [inline samples/wifi/shell/overlay-tfm.conf:29]: For a basic TF-M without Wi-Fi credentials, we don't need this change right? So, may be just rely on defaults?

Comment (simensrostad) [reply samples/wifi/shell/overlay-tfm.conf:29]: If a user applies the overlay-tfm.conf overlay my patch won't work due to the overlay kconfigs taking precedence.
How about moving all of these to the the boards/_ns files and keeping the heap size.

Then you don't have do include this overlay when building with non-secure.

Comment (krish2718) [reply samples/wifi/shell/overlay-tfm.conf:29]: Yes, that would be better.

Comment (simensrostad) [reply samples/wifi/shell/overlay-tfm.conf]: fixed, please see the changes.

File: samples/wifi/shell/boards/nrf7002dk_nrf5340_cpuapp_ns.conf [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
```
Comment (krish2718) [inline samples/wifi/shell/boards/nrf7002dk_nrf5340_cpuapp_ns.conf:8]: Should we also change `WIFI_CREDENTIALS_BACKEND` default value if building for NS `if TF-M` or equivalent?

Comment (simensrostad) [reply samples/wifi/shell/boards/nrf7002dk_nrf5340_cpuapp_ns.conf:8]: Sounds like a good idea

Comment (simensrostad) [reply samples/wifi/shell/boards/nrf7002dk_nrf5340_cpuapp_ns.conf:8]: https://github.com/nrfconnect/sdk-nrf/pull/12793 takes care of this, please have a look.

Comment (krish2718) [reply samples/wifi/shell/boards/nrf7002dk_nrf5340_cpuapp_ns.conf:8]: Sure, just approved that PR, thanks.

**Comment by simensrostad** [comment]
@krish2718 Updated the scope of the PR a bit to cleanup all samples that explicitly sets the wifi credentials backend. This now happens automatically depending on the target build.

@glarsennordic This change includes the nRF multiservice sample. Please review.

**Review by glarsennordic — APPROVED** [review]

**Review by krish2718 — APPROVED** [review]

File: samples/wifi/shell/boards/nrf7002dk_nrf5340_cpuapp_ns.conf [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+CONFIG_NET_BUF_TX_COUNT=8
+CONFIG_NET_BUF_DATA_SIZE=128
+CONFIG_NET_TC_TX_COUNT=1
+CONFIG_NET_TX_STACK_SIZE=2048
+CONFIG_NET_RX_STACK_SIZE=2048
+
+## Printing of scan results puts pressure on queues in new locking
+## design in net_mgmt. So, use a higher timeout for a crowded
+## environment.
+CONFIG_NET_MGMT_EVENT_QUEUE_TIMEOUT=5000
```
Comment (KajaKoren) [inline samples/wifi/shell/boards/nrf7002dk_nrf5340_cpuapp_ns.conf:31]: This is already set in prj.conf

**Review by rado17 — APPROVED** [review]

File: doc/nrf/libraries/networking/wifi_credentials.rst [hunk 5]
```diff
@@ -22,8 +22,8 @@ To use the Wi-Fi credentials library, enable the :kconfig:option:`CONFIG_WIFI_CR
 
 You can pick the backend using the following options:
 
-* :kconfig:option:`CONFIG_WIFI_CREDENTIALS_BACKEND_PSA`
-* :kconfig:option:`CONFIG_WIFI_CREDENTIALS_BACKEND_SETTINGS`
+* :kconfig:option:`CONFIG_WIFI_CREDENTIALS_BACKEND_PSA` - Default option for non-secure targets which includes a TF-M partition. (non-minimal TF-M profile type)
+* :kconfig:option:`CONFIG_WIFI_CREDENTIALS_BACKEND_SETTINGS` - Default option for secure targets.
```
Comment (divipillai) [inline doc/nrf/libraries/networking/wifi_credentials.rst:26]:
```suggestion
* :kconfig:option:`CONFIG_WIFI_CREDENTIALS_BACKEND_PSA` - Default option for non-secure targets, which includes a TF-M partition (non-minimal TF-M profile type).
* :kconfig:option:`CONFIG_WIFI_CREDENTIALS_BACKEND_SETTINGS` - Default option for secure targets.
```

File: samples/wifi/shell/boards/nrf7002dk_nrf5340_cpuapp_ns.conf (hunk 4 above)
Comment (simensrostad) [reply samples/wifi/shell/boards/nrf7002dk_nrf5340_cpuapp_ns.conf]: Fixed!

**Review by divipillai — APPROVED** [review]

**Review by KajaKoren — APPROVED** [review]

**Pull request merged**